<launch>
<!-- ============ Arguments ============ -->
  <arg name="scene"/>
  <arg name="model"/>
  <arg name="data_path" default="/media/blumh/data/scannet/valscans/$(arg scene)"/>
  <arg name="inference_path" default="/media/blumh/data/clustering_outputs/scannet_inference/$(arg scene)/$(arg model)"/>
  <arg name="mapname" value="pred" />
  <arg name="wait" default="false"/>
  <arg name="visualize" default="false"/>
  <arg name="namespace" default="data"/>
  <arg name="global_frame_name" default="world"/>
  <arg name="sensor_frame_name" default="depthcam"/>


<!-- ============ Play Data ============ -->
   <node pkg="panoptic_mapping_utils"
         type="pseudolabeler.py"
         name="pseudolabeler"
         output="screen"
         launch-prefix="bash -c 'sleep 2; $0 $@'"
         required="true">
    <param name="data_path" value="$(arg data_path)" />
    <param name="inference_path" value="$(arg inference_path)" />
    <param name="mapname" value="$(arg mapname)" />
    <param name="rendering_source" value="id" />
  </node>

  <node name="point_cloud_writer" pkg="point_cloud_io" type="write" output="screen">
    <param name="folder_path" value="$(arg inference_path)" />
    <param name="topic" value="/panoptic_mapper/visualization/submaps/surface_points" />
  </node>

  <!-- Mapper -->
  <node name="panoptic_mapper" pkg="panoptic_mapping_ros" type="panoptic_mapper_node" output="screen" required="true">
    <!-- Config -->
    <rosparam file="$(find panoptic_mapping_ros)/config/mapper/scannetv2_single_tsdf.yaml"/>
    <param name="save_map_path_when_finished" value=""/>

    <!-- Input -->
    <remap from="color_image_in" to="$(arg namespace)/color_image"/>
    <remap from="depth_image_in" to="$(arg namespace)/depth_image"/>
    <remap from="segmentation_image_in" to="$(arg namespace)/segmentation_image"/>
  </node>

  <!-- Map loader -->
  <node name="map_loader" pkg="panoptic_mapping_utils" type="map_loader.py" output="screen">
     <param name="path" value="$(arg inference_path)/$(arg mapname).panmap" />
     <param name="srv_name" value="/panoptic_mapper/load_map" />
     <param name="delay" value="0.1" />
  </node>

  <!-- RVIZ Visualization-->
  <node type="rviz" name="rviz" pkg="rviz" args="-d $(find panoptic_mapping_ros)/config/rviz/devel.rviz" output="screen" if="$(arg visualize)"/>

</launch>
